test/e2e_env/gatewayapi/conformance_test.go

```diff
@@ -106,8 +106,8 @@ func TestConformance(t *testing.T) {
 		RestConfig:           clientConfig,
 		Clientset:            clientset,
 		GatewayClassName:     "kuma",
-		CleanupBaseResources: true,
-		Debug:                Config.Debug,
+		CleanupBaseResources: !Config.Debug, // we need to keep the resources to collect logs when Debug is enabled
```

Comment:
gatewayapi cases are run in a dedicated CI job, so keeping these resources in the k8s cluster will not impact other cases.